### ISSUE TITLE
Provide a Java API for integration with TinkerPop3

### ISSUE THREAD

**Issue by okram** [cayleygraph/cayley#147]
If you guys provide a Java API, I will attach it to TinkerPop3 for y'all.
   http://www.tinkerpop.com/docs/3.0.0-SNAPSHOT/

-----------

As a side, is there a parallel computing notion in Cayley? That is, at minimum, be able to get "chunks of vertices". If so, then I can also provide you guys vertex-centric computing via:
  http://www.tinkerpop.com/docs/3.0.0-SNAPSHOT/#graphcomputer

Its actually pretty simple for single-machine graph systems as you can see how it is implemented in TinkerGraph (a POJO graph engine).
  https://github.com/tinkerpop/tinkerpop3/blob/master/tinkergraph-gremlin/src/main/java/com/tinkerpop/gremlin/tinkergraph/process/computer/TinkerGraphComputer.java
In other words, as long as I can get splits that I can then thread, its pretty straightforward. If I have to serially scan, its sorta wack.

**Comment by robertmeta** [comment]
This looks cool but has been sitting for awhile, moving to feature request forum: https://discourse.cayley.io/t/provide-a-java-api-for-integration-with-tinkerpop3/198
